src/auto/tls_interaction.rs

```diff
@@ -0,0 +1,127 @@
+// This file was generated by gir (https://github.com/gtk-rs/gir @ ea993ed)
+// from gir-files (https://github.com/gtk-rs/gir-files @ 469db10)
+// DO NOT EDIT
+
+use Cancellable;
+use Error;
+#[cfg(any(feature = "v2_40", feature = "dox"))]
+use TlsCertificateRequestFlags;
+#[cfg(any(feature = "v2_40", feature = "dox"))]
+use TlsConnection;
+use TlsInteractionResult;
+use TlsPassword;
+use ffi;
+use glib::object::IsA;
+use glib::translate::*;
+use glib_ffi;
+use gobject_ffi;
+use std::mem;
+use std::ptr;
+
+glib_wrapper! {
+    pub struct TlsInteraction(Object<ffi::GTlsInteraction, ffi::GTlsInteractionClass>);
+
+    match fn {
+        get_type => || ffi::g_tls_interaction_get_type(),
+    }
+}
+
+pub trait TlsInteractionExt {
+    fn ask_password<'a, P: Into<Option<&'a Cancellable>>>(&self, password: &TlsPassword, cancellable: P) -> Result<TlsInteractionResult, Error>;
+
+    fn ask_password_async<'a, P: Into<Option<&'a Cancellable>>, Q: FnOnce(Result<(), Error>) + Send + 'static>(&self, password: &TlsPassword, cancellable: P, callback: Q);
+
+    fn invoke_ask_password<'a, P: Into<Option<&'a Cancellable>>>(&self, password: &TlsPassword, cancellable: P) -> Result<TlsInteractionResult, Error>;
+
+    #[cfg(any(feature = "v2_40", feature = "dox"))]
+    fn invoke_request_certificate<'a, P: IsA<TlsConnection>, Q: Into<Option<&'a Cancellable>>>(&self, connection: &P, flags: TlsCertificateRequestFlags, cancellable: Q) -> Result<TlsInteractionResult, Error>;
+
+    #[cfg(any(feature = "v2_40", feature = "dox"))]
+    fn request_certificate<'a, P: IsA<TlsConnection>, Q: Into<Option<&'a Cancellable>>>(&self, connection: &P, flags: TlsCertificateRequestFlags, cancellable: Q) -> Result<TlsInteractionResult, Error>;
+
+    #[cfg(any(feature = "v2_40", feature = "dox"))]
+    fn request_certificate_async<'a, P: IsA<TlsConnection>, Q: Into<Option<&'a Cancellable>>, R: FnOnce(Result<(), Error>) + Send + 'static>(&self, connection: &P, flags: TlsCertificateRequestFlags, cancellable: Q, callback: R);
```

Comment:
Same problem with the async function, probably various others elsewhere too

Reply:
Done